src/functions/sendCandidateResults/application/service/send-notification.ts

```diff
@@ -0,0 +1,109 @@
+import { DocumentsServiceError } from '../../domain/errors/documents-service-error';
+import { PersonalisationDetails } from '../../domain/personalisation.model';
+import { INotifyClient } from '../../domain/notify-client.interface';
+import { get } from 'lodash';
+import { AxiosError } from 'axios';
+import * as Handlebars from 'handlebars';
+import { subjectMapper, templateMapper } from './template-selector';
+import { Language } from '../../domain/conducted-language';
+import { TestOutcome } from '../../domain/test-outcome';
+import { CommunicationMethod } from '@dvsa/mes-test-schema/categories/common';
+import { Correspondence } from '../../domain/template-id.model';
+
+export async function sendNotification(
+  emailAddress: string,
+  communicationMethod: CommunicationMethod,
+  templateId: string,
+  notificationPersonalisation: PersonalisationDetails,
+  reference: string,
+  emailReplyToId: string,
+  client: INotifyClient,
+  conductedLanguage: Language,
+  testOutcome: TestOutcome,
+  padi?: boolean,
+): Promise<any> {
+
+  try {
+    let address;
+    let renderedSubject;
+
+    let personalisation = {};
+
+    try {
+      const compileSubject = Handlebars.compile(
+        subjectMapper(notificationPersonalisation.category, conductedLanguage, padi)
+      );
+
+      renderedSubject = compileSubject(notificationPersonalisation);
+
+      personalisation = {
+        ...personalisation,
+        renderedSubject,
+      };
+    } catch (error) {
+      console.error('Error preparing renderedSubject', error);
+      return;
+    }
+
+    // add address if POST
+    if (communicationMethod === Correspondence.POST) {
+      address = {
+        address_line_1: notificationPersonalisation.address_line_1,
+        address_line_2: notificationPersonalisation.address_line_2,
+        ...(notificationPersonalisation.address_line_3 ?
+          {address_line_3: notificationPersonalisation.address_line_3} : null),
+        ...(notificationPersonalisation.address_line_4 ?
+          {address_line_4: notificationPersonalisation.address_line_4} : null),
+        ...(notificationPersonalisation.address_line_5 ?
+          {address_line_5: notificationPersonalisation.address_line_5} : null),
+        ...(notificationPersonalisation.address_line_6 ?
+          {address_line_6: notificationPersonalisation.address_line_6} : null),
+        postcode: notificationPersonalisation.postcode,
+      };
```

Comment:
Think we should clean this up and just do something like

```
import { isNil, omitBy } from 'lodash';

address = omitBy({
  address_line_1: notificationPersonalisation.address_line_1,
  address_line_2: notificationPersonalisation.address_line_2,
  address_line_3: notificationPersonalisation.address_line_3,
  address_line_4: notificationPersonalisation.address_line_4,
  address_line_5: notificationPersonalisation.address_line_5,
  address_line_6: notificationPersonalisation.address_line_6,
  postcode: notificationPersonalisation.postcode
}, isNil);
```

Reply:
like it, and tested. works nicely